packages/rum-core/src/domain/contexts/pageStateHistory.ts

```diff
@@ -38,67 +63,54 @@ export function startPageStateHistory(): PageStateHistory {
       if (!event.isTrusted) {
         return
       }
+      const startTime = event.timeStamp as RelativeTime
 
       if (event.type === DOM_EVENT.FREEZE) {
-        addPageState(PageState.FROZEN)
+        addPageState(PageState.FROZEN, startTime)
       } else if (event.type === DOM_EVENT.PAGE_HIDE) {
-        addPageState((event as PageTransitionEvent).persisted ? PageState.FROZEN : PageState.TERMINATED)
+        addPageState((event as PageTransitionEvent).persisted ? PageState.FROZEN : PageState.TERMINATED, startTime)
       } else {
-        addPageState(getPageState())
+        addPageState(getPageState(), startTime)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏what about extracting the computation of which pageState correspond to which event?
it would simplify this function